crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -396,3 +382,52 @@ impl<H: BlockHeader> BuildPendingEnv<H> for NextBlockEnvAttributes {
         }
     }
 }
+
+type BlockWithSenders<N> = Arc<RecoveredBlock<<N as NodePrimitives>::Block>>;
+type Receipts<N> = Arc<Vec<<N as NodePrimitives>::Receipt>>;
+
+/// The outcomes of building a pending block.
+#[derive(Debug)]
+pub enum PendingBlockVarious<N: NodePrimitives> {
+    /// A pending block received by the engine API from the consensus layer.
+    Consensus(Arc<RecoveredBlock<N::Block>>, Arc<Vec<N::Receipt>>),
+    /// A pending block built using the local transaction pool.
+    Pool(PendingBlock<N>),
+    /// A pending block failed to build.
+    Empty,
+}
```

Comment:
do we even need this enum?

unsure if we even need to change the local_pending_block fn